src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/VoyageAIEmbeddingClient.java

```diff
@@ -98,9 +99,10 @@ public Uni<Response> vectorize(
             .embed("Bearer " + apiKeyOverride.get(), request)
             .onFailure(
                 throwable -> {
-                  return (throwable.getCause() != null
-                      && throwable.getCause() instanceof JsonApiException jae
-                      && jae.getErrorCode() == ErrorCode.EMBEDDING_PROVIDER_TIMEOUT);
+                  return ((throwable.getCause() != null
+                          && throwable.getCause() instanceof JsonApiException jae
+                          && jae.getErrorCode() == ErrorCode.EMBEDDING_PROVIDER_TIMEOUT)
+                      || throwable instanceof TimeoutException);
```

Comment:
It's annoying to repeat the changes to every provider, and probably will have another PR to create a "BaseEmbeddingClient"